aptos-move/e2e-move-tests/src/tests/transaction_fee.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, get_stake_pool, setup_staking, transaction_fee, MoveHarness};
+use aptos_cached_packages::aptos_stdlib;
+use aptos_types::{
+    account_address::AccountAddress, on_chain_config::FeatureFlag, transaction::SignedTransaction,
+};
+
+/// Helper which creates two dummy accounts and transactions that send transfers between these accounts.
+fn p2p_txns_for_test(harness: &mut MoveHarness, num_txn_pairs: usize) -> Vec<SignedTransaction> {
+    let alice = harness.new_account_at(AccountAddress::from_hex_literal("0xa11ce").unwrap());
+    let bob = harness.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
+
+    let mut result = vec![];
+    (0..num_txn_pairs).for_each(|_| {
+        // Make transfers both ways to ensure that balances stay the same.
+        result.push(harness.create_transaction_payload(
+            &alice,
+            aptos_stdlib::aptos_coin_transfer(*bob.address(), 10),
+        ));
+        result.push(harness.create_transaction_payload(
+            &bob,
+            aptos_stdlib::aptos_coin_transfer(*alice.address(), 10),
+        ));
+    });
+    result
+}
+
+fn test_fee_collection_and_distribution(burn_percentage: u8) {
```

Comment:
Awesome! Can you also add tests for some edge cases such as:
1. The one we talked about where there's a proposal that ends the epoch.
2. Proposer leaving the validator set.
3. May be turning redistribution on and then off. Just in case we have to turn it off for some reason.
4. etc. other edge cases you can think of.

Reply:
Sounds good! I think I know how we can test all of these from the e2e suite, I will push some more changes tonight